### PR TITLE
[gui] Improve CGUIControlFactory::Create to use 'switch' instead of 'if'

### PR DESCRIPTION
**Pull request by AlwinEsch**
WIth this change the GUIControl use a switch to become related controls. This improve the code by a use of table calls instead of continues value compare.

### CODE REVIEW

File: xbmc/guilib/GUIControlFactory.cpp [hunk 1]
```diff
@@ -1103,336 +1103,368 @@ CGUIControl* CGUIControlFactory::Create(int parentID, const CRect &rect, TiXmlEl
...
-      control = new CGUILabelControl(
+      control = new CGUIFadeLabelControl(
         parentID, id, posX, posY, width, height,
-        labelInfo, wrapMultiLine, bHasPath);
-      ((CGUILabelControl *)control)->SetInfo(content);
-      ((CGUILabelControl *)control)->SetWidthControl(minWidth, (scrollValue == CGUIControl::ALWAYS) ? true : false);
+        labelInfo, scrollOut, timeToPauseAtEnd, resetOnLabelChange);
+
+      ((CGUIFadeLabelControl *)control)->SetInfo(infoLabels);
     }
```
Comment (xhaggi) [inline xbmc/guilib/GUIControlFactory.cpp]: missing break?

Comment (xhaggi) [inline xbmc/guilib/GUIControlFactory.cpp]: ignore this my mistake ;)

**Comment by xhaggi** [comment]
i've used `?w=0` to ignore whitespace changes https://github.com/xbmc/xbmc/pull/6500/files?w=0

looks good. jenkins build this please

**Comment by AlwinEsch** [comment]
jenkins build this please again.

In branch update round, I has badly also done this.

**Comment by xhaggi** [comment]
@mkortstiege or @Montellese any objections?

**Comment by mkortstiege** [comment]
Looks good and is easier to read IMO. Thanks.

**Comment by MartijnKaijser** [comment]
merge?

**Pull request merged**